js/plugins/google-cloud/src/telemetry/action.ts

```diff
@@ -42,7 +42,10 @@ class ActionTelemetry implements Telemetry {
       featureName = actionName;
     }
 
-    if (subtype === 'tool' && logInputAndOutput) {
+    if (
+      (subtype === 'tool' || actionName === 'generate') &&
```

Comment:
Are there other times where logInputAndOutput is true on different action types?